raster-foundry/raster-foundry
# Display processing imports in layer scenes UI

## Overview

This PR:
- adds a `layerId` query parameter to upload list endpoint
- inserts an upload with project default layer ID if project ID is provided but not layer ID
- makes `listItemStatus` component more generic for reuse in export list UI and in layer list UI
- adds frontend support for displaying in progress imports on V2 of layer list UI and layer scene list UI

~Marked as WIP since a small UI question needs to be addressed.~

### Checklist

- ~Styleguide updated, if necessary~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### Demo

Display import status on layer's scene list
![Screen Shot 2019-03-28 at 2 29 40 PM](https://user-images.githubusercontent.com/16109558/55183446-363d1e80-5166-11e9-9b7c-328cb0d18338.png)

Display import status on layer list
![Screen Shot 2019-03-28 at 2 30 11 PM](https://user-images.githubusercontent.com/16109558/55183397-14dc3280-5166-11e9-8b80-d04c41a62c24.png)


### Notes

This PR does not include the migration for updating the uploads table to add `defaultLayerId` to `layerId` field for those records that have `projectId` but no `layerId`. I am not sure if this is necessary... But if it is, I will include the migration in this PR.

## Testing Instructions

- `api/assembly`
- Make sure the new tests in `UploadDaoSpec` pass CI or by running `db/testOnly *UploadDaoSpec` locally
- Go to a project from V1 UI. Import a scene to this project. Grab this project ID or this upload ID. Query the database to look for this upload record. Make sure that the `layer_id` field of this record matches the project's default layer ID.
- Go to this project from V2 UI. On the layer list, make sure you see that the default layer's status field says `Processing(1)`. Go to this layer's scene list UI. Make sure that there is a card on top displaying this import status. Make sure you can close it. Make sure it shows up again if you refresh the page.
- Import another scene on this layer's scene list page. 
    * From within DB, change this new upload record's status to `FAILED`. Refresh this scene list page. Make sure that you see the status display card reflects the truth. The layer list page should reflect it as well.
    * Change both of the records to `FAILED` and refresh the page again. Make sure what is displayed still makes sense. Go back to layer list page. Make sure the status display on this layer makes sense.
- Create some exports, and manually change statuses to `FAILED`, `EXPORTED`, etc. On the export list UI, make sure the display of statuses still works

Closes https://github.com/raster-foundry/raster-foundry/issues/4682
Closes https://github.com/raster-foundry/raster-foundry/issues/4808
